x/data-proxy/keeper/keeper.go

```diff
@@ -58,3 +60,70 @@ func (k Keeper) GetDataProxyConfig(ctx context.Context, pubKey string) (result t
 
 	return config, nil
 }
+
+func (k Keeper) GetFeeUpdatePubKeys(ctx context.Context, activationHeight int64) ([][]byte, error) {
+	pubkeys := make([][]byte, 0)
+	rng := collections.NewPrefixedPairRange[int64, []byte](activationHeight)
+
+	itr, err := k.FeeUpdateQueue.Iterate(ctx, rng)
+	if err != nil {
+		return nil, err
+	}
+
+	keys, err := itr.Keys()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, k := range keys {
+		pubkeys = append(pubkeys, k.K2())
+	}
+
+	return pubkeys, nil
+}
+
+func (k Keeper) ProcessProxyFeeUpdate(ctx sdk.Context, pubKeyBytes []byte, proxyConfig *types.ProxyConfig, msg *types.MsgEditDataProxy) (int64, error) {
```

Comment:
Since this method's logic is a little dependent on `msgServer.EditDataProxy`, we can make it unexported? Also, it can take as argument a `types.FeeUpdate` struct or just `msg.FeeUpdateDelay` & `msg.NewFee`.